### PR TITLE
chore: change to vite

### PR DESCRIPTION
**Pull request by therizhao**
## Describe your changes

- CRA to vite
  - B/c CRA is very slow
  - Want to get this done before we open our repo for public contribution
- @kwokieee I changed some env vars in the files, pls check if serviceWorker part is correct

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Change REACT env vars to VITE env vars before merging to dev

### CODE REVIEW

**Review by chownces — CHANGES_REQUESTED** [review]
- can remove react-scripts dependency?
- also can we add a prepush hook to ensure the FE compiles before pushing? we lose typescript checks with vite and need to do a manual tsc to check

**Comment by CrownKira** [comment]
> * can remove react-scripts dependency?
> * also can we add a prepush hook to ensure the FE compiles before pushing? we lose typescript checks with vite and need to do a manual tsc to check

1. Agree with removing react-scripts 
2. I don't think we need a pre-push hook since CI check will fail anyways if there's type error. Should we introduce back type checking (https://github.com/fi3ework/vite-plugin-checker) since I feel it is essential in debugging but I'm not sure how much this will slow down the transpilation..

File: frontend/vite.config.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { defineConfig } from 'vite';
+import react from '@vitejs/plugin-react';
+import viteTsconfigPaths from 'vite-tsconfig-paths';
+import svgrPlugin from 'vite-plugin-svgr';
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [react(), viteTsconfigPaths(), svgrPlugin()],
+  server: {
+    port: 8080,
```
Comment (CrownKira) [inline frontend/vite.config.ts:10]: either change this to `port: 3000` or the frontend README TO `ngrok http 8080`

**Review by CrownKira — CHANGES_REQUESTED** [review]

**Review by kwokieee — CHANGES_REQUESTED** [review]
* App runs locally but does not work (Telegram login button has error displaying)
* App builds but does not work. Service worker requires some changes, as there is no more PUBLIC_URL environment variable. I am currently looking into this.
* Don't forget to update the .env.example file with the new env variables

**Comment by chownces** [comment]
> > * can remove react-scripts dependency?
> > * also can we add a prepush hook to ensure the FE compiles before pushing? we lose typescript checks with vite and need to do a manual tsc to check
> 
> 1. Agree with removing react-scripts
> 2. I don't think we need a pre-push hook since CI check will fail anyways if there's type error. Should we introduce back type checking (https://github.com/fi3ework/vite-plugin-checker) since I feel it is essential in debugging but I'm not sure how much this will slow down the transpilation..

personally i think having a hook would at least remind the user that their build is breaking before pushing. instead of finding out after CI completes. some might not check CI too, since it takes some time to run

**Comment by chownces** [comment]
> * App runs locally but does not work (Telegram login button has error displaying)
> * App builds but does not work. Service worker requires some changes, as there is no more PUBLIC_URL environment variable. I am currently looking into this.
> * Don't forget to update the .env.example file with the new env variables

agree that there are build errors now. refer to:
- `npx vite build --mode development` to check the actual minified errors
- issue 2139 on the vitejs repo (not tagging it here on purpose)

File: frontend/vite.config.ts (hunk 1 above)
Comment (therizhao) [reply frontend/vite.config.ts:10]: ok! i realise my original env port was wrong lol

**Comment by therizhao** [comment]
Remove react-scripts & add vite-plugin-checker

TODO
- [ ] Fix service worker build

**Comment by therizhao** [comment]
Removed react-telegram-login, b/c it doesn't work with Vite. Added custom implementation.

**Comment by chownces** [comment]
> Removed react-telegram-login, b/c it doesn't work with Vite. Added custom implementation.

i might write a custom package to add customizable telegram login buttons sometime in dec. seems like quite a requested usecase, and the iframe injection method telegram takes is actually preventing people from adopting it...

**Comment by therizhao** [comment]
Ok folks it turned out we need one more line
image
https://vite-pwa-org.netlify.app/guide/inject-manifest.html#service-worker-code-1 See the docs.

Tested and working. @kwokieee and everyone else try it out pls 


1. `yarn devbuild && yarn serve`
2. Make some changes to the UI code
3. `yarn devbuild && yarn serve`
SHOULD UPDATE PROPERLY WITHOUT 404 errors

**Comment by kwokieee** [comment]
image
https://vite-pwa-org.netlify.app/guide/faq.html#missing-assets-from-sw-precache-manifest
Made one small change to the maximum file size that the service worker can precache. Now should be working properly.

**Review by kwokieee — APPROVED** [review]

**Comment by therizhao** [comment]
> 
> 
> image
> 
> https://vite-pwa-org.netlify.app/guide/faq.html#missing-assets-from-sw-precache-manifest
> 
> Made one small change to the maximum file size that the service worker can precache. Now should be working properly.

Btw why limit to 4MB? Instead of unlimited (1GB or something)

**Comment by kwokieee** [comment]
> > image
> > https://vite-pwa-org.netlify.app/guide/faq.html#missing-assets-from-sw-precache-manifest
> > Made one small change to the maximum file size that the service worker can precache. Now should be working properly.
> 
> Btw why limit to 4MB? Instead of unlimited (1GB or something)

We shouldn't set the limit to a large value, if not people will use a lot of data when accessing our website. Anyways, our largest file being precached now is about 2.78MB, so still got plenty of room (default value is 2MB)

**Pull request merged**